functions/src/definitions/common/repostUtils.ts

```diff
@@ -0,0 +1,133 @@
+/*
+Stores a collection of functions for reposting messages to a admin channel
+*/
+import { AppEnv } from "../../appEnv"
+import {
+  sendTelegramImageMessage,
+  sendTelegramTextMessage,
+  updateTelegramTextMessage,
+  updateTelegramImageMessage,
+} from "./sendTelegramMessage"
+import * as admin from "firebase-admin"
+import { getSignedUrl } from "./mediaUtils"
+
+const db = admin.firestore()
+const repostChannelId = process.env[AppEnv.TELEGRAM_REPOST_CHANNEL_ID] || ""
+
+// Stores the replyId and instanceText whenever we repost a message
+export async function createMessageId(
+  messageId: string,
+  replyId: string,
+  instanceText: string
+) {
+  try {
+    const repostIdRef = db.collection("repostIds").doc(messageId)
+    await repostIdRef.set({ replyId: replyId, instanceText: instanceText })
+  } catch (error) {
+    console.error("repostUtils > Error adding document: ", error)
+    return false
+  }
+  return true
+}
+
+// Retrieves the replyId and instanceText for a given messageId to reply/update message
+export async function getReplyId(messageId: string) {
+  try {
+    const repostIdRef = db.collection("repostIds").doc(messageId)
+    const repostIdSnap = await repostIdRef.get()
+
+    if (!repostIdSnap.exists) {
+      throw new Error(`Repost ID for messageId ${messageId} not found`)
+    }
+
+    const data = repostIdSnap.data()
+    return {
+      replyId: data?.replyId || null,
+      instanceText: data?.instanceText || null,
+    }
+  } catch (error) {
+    console.error("Error getting repost ID", error)
+    throw error
+  }
+}
+
+// Creates a new text message in the repost channel
+export async function repostText(messageId: string, instance: any) {
```

Comment:
messageId here should be the ID of the firestore document in the Messages collection. Currently, it's the whatsapp message ID that is tied to the instance.